src/number/streaming.rs

```diff
@@ -1578,6 +1663,79 @@ where
   Ok((i, float))
 }
 
+/// Recognizes floating point number in text format and returns a f64.
+/// This only handles non-finite (special) values.
+pub fn double_nonfinite<T, E: ParseError<T>>(input: T) -> IResult<T, f64, E>
+where
+  T: Slice<RangeFrom<usize>> + Slice<RangeTo<usize>>,
+  T: Clone + Offset,
+  T: InputIter + InputLength + InputTake + crate::traits::ParseTo<i32>,
+  <T as InputIter>::Item: AsChar,
+  <T as InputIter>::IterElem: Clone,
+  T: InputTakeAtPosition,
+  <T as InputTakeAtPosition>::Item: AsChar,
+  T: AsBytes,
+  T: for<'a> Compare<&'a [u8]>,
+{
+  let (i, sign) = sign(input.clone())?;
+  let (mut double, count) = if case_insensitive_cmp(i.as_bytes(), b"nan") {
+    (f64::NAN, 3)
+  } else if i.input_len() >= 8 && case_insensitive_cmp(i.as_bytes(), b"infinity") {
```

Comment:
`infinity` is not in the EBNF of rust std

Reply:
But in practice it is https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=82d184db3092ef5e95cb62f2a174871d

Reply:
I starting to doubt about the doc of std be up to date

Reply:
> `infinity` is not in the EBNF of rust std

It is, actually. See below:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d5ac6a9ba62860ebe04f782970f95342

Reply:
Ah I found the PR that fixed this for IEEE-754 compliance:
https://github.com/rust-lang/rust/pull/78618

I'm not sure when this made it into stable, however, and it was definitely after 1.51, so it won't compare to the core implementation.